controllers/storagecluster/blackbox_exporter.go

```diff
@@ -0,0 +1,563 @@
+	}
+	actual := &securityv1.SecurityContextConstraints{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: desired.Name,
+		},
+	}
+
+	err := r.Client.Get(ctx, types.NamespacedName{Name: actual.Name}, actual)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			if err := r.Client.Create(ctx, desired); err != nil {
+				r.Log.Error(err, "Failed to create SCC", "SCC", desired.Name)
+				return err
+			}
+			r.Log.Info("Created SCC", "SCC", desired.Name)
+			return nil
+		}
+		r.Log.Error(err, "Failed to get SCC", "SCC", desired.Name)
+		return err
+	}
+
+	// Only update if the labels differ
+	if !equality.Semantic.DeepEqual(desired.Labels, actual.Labels) {
+
+		actual.Labels = desired.Labels
+		if err := controllerutil.SetControllerReference(instance, actual, r.Scheme); err != nil {
+			r.Log.Error(err, "Failed to set owner reference on SCC", "SCC", actual.Name)
+			return err
+		}
+
+		// Preserve existing users/groups
+		actual.Users = mergeStringSlices(actual.Users, desired.Users)
+		actual.Groups = mergeStringSlices(actual.Groups, desired.Groups)
+
+		if err := r.Client.Update(ctx, actual); err != nil {
+			r.Log.Error(err, "Failed to update SCC", "SCC", actual.Name)
+			return err
+		}
+		r.Log.Info("Updated SCC", "SCC", actual.Name)
+	}
+
+	return nil
+}
+
+func mergeStringSlices(existing, desired []string) []string {
+	result := make([]string, len(existing))
+	copy(result, existing)
+	for _, item := range desired {
+		if !slices.Contains(result, item) {
+			result = append(result, item)
+		}
+	}
+	return result
+}
+
+// createBlackboxConfigMap creates or updates the ConfigMap
+func (r *StorageClusterReconciler) createBlackboxConfigMap(ctx context.Context, instance *ocsv1.StorageCluster) error {
+
+	desired := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      blackboxConfigMapName,
+			Namespace: instance.Namespace,
+			Labels:    blackboxExporterLabels,
+		},
+		Data: map[string]string{
+			"config.yml": `
+modules:
+  icmp_internal:
+    prober: icmp
+    timeout: 5s
+    icmp:
+      preferred_ip_protocol: ip4
+`,
+		},
+	}
+
+	actual := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      desired.Name,
+			Namespace: desired.Namespace,
+		},
+	}
+
+	_, err := controllerutil.CreateOrUpdate(ctx, r.Client, actual, func() error {
+		if actual.CreationTimestamp.IsZero() {
+			actual.Labels = desired.Labels
+			if err := controllerutil.SetControllerReference(instance, actual, r.Scheme); err != nil {
+				return err
+			}
+		}
+		actual.Data = desired.Data
+		return nil
+	})
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		r.Log.Error(err, "Failed to create/update ConfigMap for Blackbox Exporter")
+		return err
+	}
+
+	r.Log.Info("ConfigMap for Blackbox Exporter is ready")
+	return nil
+}
+
+func (r *StorageClusterReconciler) getNodeIPs(ctx context.Context) ([]string, error) {
+	nodes := &corev1.NodeList{}
+	err := r.Client.List(ctx, nodes, client.MatchingLabels{defaults.NodeAffinityKey: ""})
```

Comment:
do we target nodes which do not have a label?

Reply:
no, we shouldn't